services/m3dbnode/server/server.go

```diff
@@ -226,38 +234,80 @@ func Run(runOpts RunOptions) {
+					SetLogger(logger).
+					SetMetricsScope(scope))
+		configSvcClient, err := etcdclient.NewConfigServiceClient(configSvcClientOpts)
+		if err != nil {
+			logger.Fatalf("could not create m3cluster client: %v", err)
+		}
 
-	dynamicOpts := namespace.NewDynamicOptions().
-		SetInstrumentOptions(iopts).
-		SetConfigServiceClient(configSvcClient).
-		SetNamespaceRegistryKey(kvconfig.NamespacesKey)
-	nsInit := namespace.NewDynamicInitializer(dynamicOpts)
+		dynamicOpts := namespace.NewDynamicOptions().
+			SetInstrumentOptions(iopts).
+			SetConfigServiceClient(configSvcClient).
+			SetNamespaceRegistryKey(kvconfig.NamespacesKey)
+		nsInit := namespace.NewDynamicInitializer(dynamicOpts)
 
-	opts = opts.SetNamespaceInitializer(nsInit)
+		opts = opts.SetNamespaceInitializer(nsInit)
 
-	serviceID := services.NewServiceID().
-		SetName(cfg.ConfigService.Service).
-		SetEnvironment(cfg.ConfigService.Env).
-		SetZone(cfg.ConfigService.Zone)
+		serviceID := services.NewServiceID().
+			SetName(cfg.ConfigService.Service).
+			SetEnvironment(cfg.ConfigService.Env).
+			SetZone(cfg.ConfigService.Zone)
 
-	topoOpts := topology.NewDynamicOptions().
-		SetConfigServiceClient(configSvcClient).
-		SetServiceID(serviceID).
-		SetQueryOptions(services.NewQueryOptions().SetIncludeUnhealthy(true)).
-		SetInstrumentOptions(opts.InstrumentOptions()).
-		SetHashGen(sharding.NewHashGenWithSeed(cfg.HashingConfiguration.Seed))
+		topoOpts := topology.NewDynamicOptions().
+			SetConfigServiceClient(configSvcClient).
+			SetServiceID(serviceID).
+			SetQueryOptions(services.NewQueryOptions().SetIncludeUnhealthy(true)).
+			SetInstrumentOptions(opts.InstrumentOptions()).
+			SetHashGen(sharding.NewHashGenWithSeed(cfg.HashingConfiguration.Seed))
+
+		topoInit = topology.NewDynamicInitializer(topoOpts)
+
+		kv, err = configSvcClient.KV()
+		if err != nil {
+			logger.Fatalf("could not create KV client, %v", err)
+		}
+
+	case cfg.StaticConfig != nil && cfg.StaticConfig.TopologyConfig != nil && cfg.StaticConfig.Namespaces != nil:
+		logger.Info("creating static config service client with m3cluster")
+
+		shardSet, hostShardSets, err := createStaticShardSet(cfg.StaticConfig.TopologyConfig.Shards, cfg.ListenAddress)
+		if err != nil {
+			logger.Fatalf("unable to create shard set for static config: %v", err)
+		}
+		staticOptions := topology.NewStaticOptions().
+			SetReplicas(1).
+			SetHostShardSets(hostShardSets).
+			SetShardSet(shardSet)
+
+		nsList := []namespace.Metadata{}
+		for _, ns := range cfg.StaticConfig.Namespaces {
+			md, err := createMetaData(ns)
+			if err != nil {
+				logger.Fatalf("unable to create metadata for static config: %v", err)
+			}
+			nsList = append(nsList, md)
+		}
+		nsInitStatic := namespace.NewStaticInitializer(nsList)
+		topoInit = topology.NewStaticInitializer(staticOptions)
+		opts = opts.SetNamespaceInitializer(nsInitStatic)
+
+		kv = m3clusterkvmem.NewStore()
+
+	default:
+		logger.Fatal("configService or staticConfigService required")
```

Comment:
This looks outdated, should be `config service or static configuration required` perhaps?